spec/java/util/ArrayList.lsl

```diff
@@ -0,0 +1,1205 @@
+        //    ): list<any>;
+
+        //But Collection isn't List.
+        //But it has method: toArray();
+        //Maybe we must have two arrays as params ?
+
+        //storage = action INTERSECTION(storage, c);
+
+        action NOT_IMPLEMENTED();
+    }
+
+
+    @private
+    @throws(["java.io.IOException"])
+    fun writeObject (s: ObjectOutputStream): void
+    {
+        var expectedModCount = modCount;
+
+        // #problem
+        //Here is cycle and some operations with ObjectOutputStream.
+        action NOT_IMPLEMENTED();
+
+        if (modCount != expectedModCount)
+        {
+            action THROW_NEW("java.util.ConcurrentModificationException", []);
+        }
+    }
+
+
+    @private
+    @throws(["java.io.IOException", "java.lang.ClassNotFoundException"])
+    fun readObject (s: ObjectInputStream): void
+    {
+        // #problem
+        //Here is cycle and some operations with ObjectInputStream.
+        action NOT_IMPLEMENTED();
+    }
+
+
+    fun listIterator (index: int): ListIterator
+    {
+        _rangeCheckForAdd(index, length);
+
+        result = new ListItr(state=Created,
+            cursor=index,
+            expectedModCount=modCount);
+    }
+
+
+    fun listIterator (): ListIterator
+    {
+        result = new ListItr(state=Created,
+            cursor=0,
+            expectedModCount=modCount);
+    }
+
+
+    fun iterator (): Iterator
+    {
+        result = new Itr(state=Created,
+            expectedModCount=modCount);
+    }
+
+
+    fun subList (fromIndex: int, toIndex: int): List
+    {
+        _subListRangeCheck(fromIndex, toIndex, length);
+
+        // #problem
+        //We don't have decision about sublists.
+        //TODO
+        result = new SubList(state=Created,
+            startIndex=fromIndex,
+            endIndex=toIndex);
+    }
+
+
+    fun forEach (action: Consumer): void
+    {
+        // #problem
+        action NOT_IMPLEMENTED();
+    }
+
+
+    fun spliterator (): Spliterator
+    {
+        result = new ArrayListSpliterator(state=Initialized,
+            //This is right ? "parent=self"
+            parent=self,
+            origin = 0,
+            est=-1,
+            expectedModCount=0);
+    }
+
+
+    fun removeIf (filter: Predicate): boolean
+    {
+        // #problem
+        //We don't know how to work with lambda.
+        action NOT_IMPLEMENTED();
+    }
+
+
+    fun replaceAll (operator: UnaryOperator): void
+    {
+        // #problem
+        action NOT_IMPLEMENTED();
+    }
+
+
+    fun sort (c: Comparator): void
+    {
+        var expectedModCount = modCount;
+
+        // #problem
+        //We don't know what to do with sorting of the List.
+        action NOT_IMPLEMENTED();
+
+        if (modCount != expectedModCount)
+        {
+            action THROW_NEW("java.util.ConcurrentModificationException", []);
+        }
+
+        modCount = modCount + 1;
+    }
+}
+
+
+
+@packagePrivate
+@implements("java.util.Iterator")
+@WrapperMeta(
```

Comment:
This should not be a wrapper as it is just an internal/private class.

Reply:
Fixed